pkg/helm/helm_helpers.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"regexp"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/log"
+)
+
+type k8sObj struct {
+	ObjKind  string            `json:"kind"`
+	MetaData metav1.ObjectMeta `json:"metadata"`
+}
+
+type K8sObjectType string
+
+type Component struct {
+	k8sType K8sObjectType
+	name    string
+}
+
+/*
+	This file contains utility functions primarily to be used to help write test cases.
+	These are to be used in conjunction with helm install --dry-run mode.
+	The output from dry run will be sent to each of these functions that would generate either `release` name or slice
+	of `Component`struct that help to run tests and validate.
+*/
+
+// ParseReleaseNameFromHelmStatus takes as input the rendered output from a dry-run enabled HelmApp Install and tries to
+// extract the release name from it for validation.
+func ParseReleaseNameFromHelmStatus(helmStatus string) string {
```

Comment:
We are getting release name form renderedOutput right? so should we not call it `ReleaseNameFromRenderedOutput`?
Also, the comment says we extract the release name from rendered output for validation, can you explain a bit more, which validation are we talking about?
When we run `helm install` do we not know the release name already?

Reply:
This method was already present. I added test coverage to it. As of now in this PR it is not used. But I think this was written with the purpose of reference in case we would like to compare some other attribute, other than `release name`.
Do let me know if this utility function be removed in case you have different opinion.

Reply:
Removed this func as it is not used in any tests